### PR TITLE
[pulsar-client-cpp] fix snappy compressor compile error

### LINKED ISSUE #4720: Compilation errors in `pulsar-client-cpp`
**Describe the bug**
When attempting to compile `pulsar-client-cpp`, I get these compilation errors:

```
[  7%] Building CXX object lib/CMakeFiles/pulsarStatic.dir/CompressionCodecSnappy.cc.o                                                                                                                                                         
/home/mediocregopher/src/pulsar/pulsar-client-cpp/lib/CompressionCodecSnappy.cc: In member function ‘virtual pulsar::SharedBuffer pulsar::CompressionCodecSnappy::encode(const pulsar::SharedBuffer&)’:                                     
/home/mediocregopher/src/pulsar/pulsar-client-cpp/lib/CompressionCodecSnappy.cc:38:59: error: ‘res’ was not declared in this scope
   38 |         LOG_ERROR("Failed to compress to snappy. res=" << res);
      |                                                           ^~~
/home/mediocregopher/src/pulsar/pulsar-client-cpp/lib/CompressionCodecSnappy.cc:38:9: error: ‘LOG_ERROR’ was not declared in this scope; did you mean ‘SO_ERROR’?
   38 |         LOG_ERROR("Failed to compress to snappy. res=" << res);
      |         ^~~~~~~~~
      |         SO_ERROR
/home/mediocregopher/src/pulsar/pulsar-client-cpp/lib/CompressionCodecSnappy.cc: In member function ‘virtual bool pulsar::CompressionCodecSnappy::decode(const pulsar::SharedBuffer&, uint32_t, pulsar::SharedBuffer&)’:
/home/mediocregopher/src/pulsar/pulsar-client-cpp/lib/CompressionCodecSnappy.cc:52:74: error: invalid conversion from ‘uint32_t’ {aka ‘unsigned int’} to ‘size_t*’ {aka ‘long unsigned int*’} [-fpermissive]
   52 |                                              decompressed.mutableData(), uncompressedSize);
      |                                                                          ^~~~~~~~~~~~~~~~
      |                                                                          |
      |                                                                          uint32_t {aka unsigned int}
In file included from /home/mediocregopher/src/pulsar/pulsar-client-cpp/lib/CompressionCodecSnappy.cc:23:
/usr/include/snappy-c.h:106:41: note:   initializing argument 4 of ‘snappy_status snappy_uncompress(const char*, size_t, char*, size_t*)’
  106 |                                 size_t* uncompressed_length);
      |                                 ~~~~~~~~^~~~~~~~~~~~~~~~~~~
```

**To Reproduce**
```
git clone https://github.com/apache/pulsar.git
# git checkout v2.4.0 (optional)
cd pulsar/pulsar-client-cpp
cmake .
make
```
As shown, this happens on both the `v2.4.0` tag and `master`

**Expected behavior**
The build successfully finishes.

**Desktop (please complete the following information):**
```
 ~ :: uname -a
Linux very-cool-thinkpad 5.1.16-arch1-1-ARCH #1 SMP PREEMPT Wed Jul 3 20:23:07 UTC 2019 x86_64 GNU/Linux

 ~ :: pacman -Q | grep boost
boost 1.69.0-2
boost-libs 1.69.0-2

 ~ :: pacman -Q | grep proto
protobuf 3.7.0-1

 ~ :: pacman -Q | grep log4cxx
log4cxx 0.10.0-4

 ~ :: pacman -Q | grep curl
curl 7.65.1-3
libcurl-compat 7.65.1-1

 ~ :: pacman -Q | grep gtest
gtest 1.8.1-3

 ~ :: pacman -Q | grep jsoncpp
jsoncpp 1.8.4-2

 ~ :: pacman -Q | grep clang
clang 8.0.0-4

 ~ :: pacman -Q | grep glibc
glibc 2.29-3
```

**Additional context**
I am able to build the library files using the `docker-build.sh` script, but the built files aren't usable on my system because they are not statically compiled, and the versions of libraries they link to are different.

### PR DESCRIPTION
**Pull request by yittg**
Fixes #4720 
### Motivation

for now, the snappy compressor can not be compiled successfully,
which raise undeclared identifier and no matching function errors.

### Modifications

fix compile errors, use snappy c++ apis.

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

`libsnappy-dev` is installed from Dockerfile, but cmake got `HAS_SNAPPY: 0` in CI as outdated docker image, so these codes were not compiled ever in CI.
May we checkout a new build image, `apachepulsar/pulsar-build:ubuntu-16.04` has no `libsnappy` installed.

### CODE REVIEW

**Comment by sijie** [comment]
run cpp tests

**Comment by sijie** [comment]
run java8 tests

**Comment by sijie** [comment]
run java8 tests

**Comment by yittg** [comment]
run java8 tests

File: pulsar-client-cpp/docker-lib-check.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+if ! dpkg -l libsnappy-dev; then
+    apt-get install -y libsnappy-dev
```
Comment (sijie) [inline pulsar-client-cpp/docker-lib-check.sh]: @yittg can you add the library to https://github.com/apache/pulsar/blob/master/build/docker/Dockerfile

Comment (yittg) [reply pulsar-client-cpp/docker-lib-check.sh]: As I said, it's in [Dockerfile](https://github.com/apache/pulsar/blob/41bc407e1642aec3805d0f077694edd835b3ea1c/build/docker/Dockerfile#L31), but the docker image with tag `ubuntu-16.04` is outdated which has no full libs installed.

**Pull request merged**